### ISSUE TITLE
Crash when doing local stun request - is this expected?

### ISSUE THREAD

**Issue by marcbrevoort-cyberhive** [sile/rustun#5]
Scenario:
- Running stun server with ```cargo run --example binding_srv```
- clone python stun client from git@github.com:arantonitis/pynat.git (run under python 3)
- Connect to stun server with this client through `python pynat.py  --stun-host 0.0.0.0`
- Result:
```

Error: Other (cause; assertion failed: `left == right`; assertion failed: `(left == right)` (left: `1180954873`, right: `554869826`))
HISTORY:
  [0] at /home/mrjb/.cargo/registry/src/github.com-1ecc6299db9ec823/stun_codec-0.1.12/src/message.rs:287
  [1] at /home/mrjb/.cargo/registry/src/github.com-1ecc6299db9ec823/bytecodec-0.4.13/src/combinator.rs:1369
  [2] at /home/mrjb/.cargo/registry/src/github.com-1ecc6299db9ec823/stun_codec-0.1.12/src/message.rs:424
  [3] at /home/mrjb/.cargo/registry/src/github.com-1ecc6299db9ec823/bytecodec-0.4.13/src/decode.rs:449
  [4] at /home/mrjb/.cargo/registry/src/github.com-1ecc6299db9ec823/fibers_transport-0.1.2/src/udp.rs:200
  [5] at /home/mrjb/.cargo/registry/src/github.com-1ecc6299db9ec823/fibers_transport-0.1.2/src/error.rs:22 -- original_error_kind=InvalidInput
  [6] at src/transport/udp.rs:358
  [7] at src/transport/udp.rs:193
  [8] at src/channel.rs:178
  [9] at src/error.rs:63 -- original_error_kind=CodecError
  [10] at src/server.rs:331
  [11] at src/server.rs:71
  [12] at examples/binding_srv.rs:32
```
Granted it would be weird to run a stun server and access it from a local client - is this expected?

**Comment by sile** [comment]
I took a look at the `pynat` README.md and found that it says "PyNAT follows RFC 3489".
RFC 3489 is obsoleted by RFC 5389 and `rustun` only support RFC 5389. I think the above error is due to the protocol version mismatch.

**Comment by marcbrevoort-cyberhive** [comment]
Indeed, I confirm that if I do a STUN request with `go-stun` which is RFC-5389 enabled, this crash does not occur and I get a sensible reply. Perhaps incorrect data should just be rejected to prevent denial-of-service attacks to be (accidentally) carried out in this manner?

**Comment by sile** [comment]
> incorrect data should just be rejected to prevent denial-of-service attacks ...

I think so. `binding_srv.rs` is simply aborted when it receives incorrect data because is just an example command. But in real applications, such errors should be properly handled without aborting.

**Comment by sile** [comment]
On the second thought, I think that the above behavior (i.e., aborting the program) may be an unexpected one. I'm going to read the implementation code and fix the problem if need.

**Comment by sile** [comment]
I addressed the problem at #6 .

**Comment by sile** [comment]
Let me close this issue as the original question seems to have been answered.
